src/v/cluster/metadata_dissemination_service.cc

```diff
@@ -343,7 +343,7 @@ ss::future<> metadata_dissemination_service::dispatch_disseminate_leadership() {
     return _health_monitor.local()
       .get_cluster_health(
         cluster_report_filter{},
-        force_refresh::no,
+        force_refresh::yes,
```

Comment:
I haven't walked through how difficult the plumbing is for the second option you mentioned would be, but an unconditional refresh seems a bit scary.

Another idea is to plumb the `health_monitor_backend` into the `controller_backend` and introduce a call to the `health_monitor_backend` that resets its `_last_refresh` time, so we know to refresh metadata when we delete a partition, but not in the steady state.

Generally I wonder if there's a finer grained way to make the health monitor forget about specific partitions, but I suspect not. Maybe something we should improve in the future.

Reply:
Yeah, it's an unfortunate change really. I thought about it some more and I don't think the second option (bumping the term before deletion) would actually work. If the entry is missing from the partition leaders table, it will get added regardless of the term.

Reply:
I wouldn't do this, as this way we make the health monitor artificially updated. Maybe it would be better to consult the `topics_table` before adding leader to leader table. i.e. add it only if partition exists ?

Reply:
I like that idea. `partition_leaders_table` conveniently has a ref to the topic table.

Reply:
That approach didn't work as it prevented insertions for newly created topics. I came up with a different approach: the health monitor now provides a hint which indicates if health metadata was refreshed. The metadata dissemination service uses the hint to avoid applying leader updates with stale metadata.